iOS/ExposureNotification/source/ApiDefinition.cs

```diff
@@ -367,17 +352,19 @@ interface ENManager {
 		[Export ("exposureNotificationStatus", ArgumentSemantic.Assign)]
 		ENStatus ExposureNotificationStatus { get; }
 
-		[NullAllowed, Export("invalidationHandler", ArgumentSemantic.Copy)]
+		[NullAllowed, Export ("invalidationHandler", ArgumentSemantic.Copy)]
 		Action InvalidationHandler { get; set; }
 
+		[NullAllowed, Export ("activityHandler", ArgumentSemantic.Copy)]
+		Action<ENActivityFlags> LaunchActivityHandler { get; set; }
```

Comment:
@dalexsoto is this the equivalent code for setting "value for key"? I know that there is a `SetValueForKey`, but I was hoping to leverage the binding engine...

```objc
typedef NS_OPTIONS( uint32_t, ENActivityFlags )
{
    /// The app launched to perform periodic operations.
    ENActivityFlagsPeriodicRun = ( 1U << 2 )
};

typedef void (^ENActivityHandler)(ENActivityFlags activityFlags);

@interface ENManager (ENActivityHandling)

@end

@implementation ENManager (ENActivityHandling)

- (void)setLaunchActivityHandler:(ENActivityHandler)activityHandler {
    [self setValue:activityHandler forKey:@"activityHandler"];
}

@end
```

Reply:
if the implementation only in the headers ?
in that case it might not be in the binary so you can't bind it
but you can provide a (manually written) method that does the same
you might be able to use `[Wrap("...")]` but that does not debug well and is not verified by the compiler (well it is after code is generated)

Reply:
How does one properly wrap a `Action<T>` to `NSObject`? Is there a clean wrapper?

The sample usage was written like this:

```objc
// Set the launch activity handler.
[manager setLaunchActivityHandler:^(ENActivityFlags activityFlags) {
    if (activityFlags & ENActivityFlagsPeriodicRun) {
        // Your app now has 3.5 minutes to perform download and detection.
    }
}];
```

Reply:
Also, the implementation is recommended to inject it yourself. Seems this value is something that only the older implementations look at. I think you are literally supposed to just set this from outside, but you wrap it in the "extensions" for niceness

```objc
[manager setValue:activityHandler forKey:@"activityHandler"];
```

Reply:
This may require you to create the obj-c block from a managed delegate yourself kinda like show in this file 

https://github.com/xamarin/xamarin-macios/blob/6a6f1f1ff8256e154acca6519d8bc56553e25845/src/Network/NWFramer.cs#L129-L147

But instead of passing the block pointer to a c function as shown above you may need to add this member to `ENManager` and give the block pointer to that function

```
[Sealed]
[Export ("setValue:forKey:")]
void SetValueForKey (IntPtr ptr, NSString key);
```

This all depends on how/who calls into the the `activityHandler` block, if it is the native code then you have no other choice than doing this manually, if it is you from your C# code then you can avoid all of this and stay in managed land.

@spouliot do you agree? I hope I am not posting non-sense at 6 pm :P

Reply:
@dalexsoto agreed, with some ☕ to back it up 😄